### ISSUE TITLE
Provide Preview Service

### ISSUE THREAD

**Issue by b2m** [gitonthescene/csv-reconcile#28]
I am reconciling against local CSV files with ambiguous data in the search column and without ids from external systems (Wikidata...).

| ID | Search Column  | Additional Data |
| -- |----------------- | ---------------- |
| 1  | John Doe           | 1970-01-01      |
| 2  | John Doe           | 2020-01-01      |
| ...  | ...                       | ...                       |

To identify the correct match from the Reconciliation API I have to manually check the proposed results against the data in the CSV files. To speed this process up I would prefer to use a Preview Service as defined in the [Reconciliation Service API Specification](https://reconciliation-api.github.io/specs/latest/#preview-service).

**Comment by gitonthescene** [comment]
Oohh.. I noticed that when the service is shut down then hovering over it shows "unable to connect". Since these are imagined to be temporary services, that's kind of annoying. Also, since you may be making many temporary services running on the same port the preview service may not make sense if it's showing data from the current service.

I'm not sure what the right thing to do is. What do you think about not having this on by default? You can always add it by supplying a config to point to it. This sort of side steps the issue until we come up with a better idea.

**Comment by b2m** [comment]
Yes I see the problem (literally and figuratively).

So one problem is, that we (may) use the exact same service to reconcile against many different databases (CSV-files).
One way to adress this problem would be to have different ports for each service and somehow handle them without getting confused =)

Another idea to adress this issue is to add a layer to the reconcilation service to be able to provide reconciliation endpoints for multiple databases (csv files) at once.

Like:
 - http://localhost:5000/persons_csv/{reconcile|preview|...}
 - http://localhost:5000/places_csv/{reconcile|preview|...}

This would also require some CLI or interface logic to remove a database, as they are then somehow persistent.

Not sure whether this is a path we want to follow along, because I do not know the use-cases of the other users of this library.

For me a local reconciliation service is currently a "use & forget" affair, because I am doing a lot of unique file transformations.

But I see the point of having some local databases (csv files) that I may use together or multiple times.

----------

**Conclusion:** For me it would be OK to have the preview service as optional feature, when it is anoying other users.
 If you think that handling multiple reconciliation services at once is a nice improvement, then let us open a ticket and discuss/implement it.

**Comment by gitonthescene** [comment]
Sorry I took so long to get to this, but this is done now.  Thinking about it a little bit more.  It might be nice if OpenRefine checked the service manifests periodically and disabled the links for downed services.  [This](https://github.com/OpenRefine/OpenRefine/issues/3645) is a related issue.

FWIW, I made another small change to the interface to relieve users from having to remember the same options for the serve step that they passed to the init step.  Now the serve step takes no options but uses configs saved off during the init step.
